src/others/helpers/use_query.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo, useState } from "react";
+import { FilterItemId, FilterName } from "../contexts/filter";
+
+type QueryName = "category" | "city" | "date";
+
+const queryNametoFilterMap: {
+  [key in FilterName]: QueryName;
+} = {
+  Categories: "category",
+  Cities: "city",
+  Dates: "date",
+};
+
+export const useQuery = (
+  search: string,
```

Comment:
**suggestion/naming**

How about calling it a `urlParamsString`? I think it's less ambiguous.